automation/release.sh

```diff
@@ -0,0 +1,90 @@
+    if command -V gh; then
+        return
+    fi
+
+    trap 'cleanup_gh_install' EXIT SIGINT SIGTERM
+
+    # install gh cli for uploading release artifacts, with prompt disabled to enforce non-interactive mode
+    gh_cli_dir=$(mktemp -d)
+    (
+        cd  "$gh_cli_dir/"
+        curl -sSL "https://github.com/cli/cli/releases/download/v${GH_CLI_VERSION}/gh_${GH_CLI_VERSION}_linux_amd64.tar.gz" -o "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+        tar xvf "gh_${GH_CLI_VERSION}_linux_amd64.tar.gz"
+    )
+    export PATH="$gh_cli_dir/gh_${GH_CLI_VERSION}_linux_amd64/bin:$PATH"
+    if ! command -V gh; then
+        echo "gh cli not installed successfully"
+        exit 1
+    fi
+    gh config set prompt disabled
+}
+
+function main() {
+
+    DOCKER_TAG="$(git tag --points-at HEAD | head -1)"
+    if [ -z "$DOCKER_TAG" ]; then
+        echo "commit $(git show -s --format=%h) doesn't have a tag, exiting..."
+        exit 0
+    fi
+    export DOCKER_TAG
+    export TRAVIS_TAG="$DOCKER_TAG"
+
+    ensure_gh_cli_installed
+
+    make
+    make build-verify # verify that we set version on the packages built by go (goveralls depends on go-build target)
+    make apidocs
+    make client-python
+    make manifests DOCKER_PREFIX="$DOCKER_PREFIX" DOCKER_TAG="$DOCKER_TAG" # skip getting old CSVs here (no QUAY_REPOSITORY), verification might fail because of stricter rules over time; falls back to latest if not on a tag
+    make olm-verify
+    make prom-rules-verify
+
+    make manifests
+    make build-functests
+
+    # add artifacts to created github release (overwrite existing)
+    gh auth login --with-token <"$GITHUB_TOKEN_PATH"
+    gh release upload --clobber "$DOCKER_TAG" _out/cmd/virtctl/virtctl-v* \
+        _out/manifests/release/demo-content.yaml \
+        _out/manifests/release/kubevirt-operator.yaml \
+        _out/manifests/release/kubevirt-cr.yaml \
+        _out/templates/manifests/release/kubevirt-operator.yaml.j2 \
+        _out/templates/manifests/release/kubevirt-cr.yaml.j2 \
+        _out/templates/manifests/release/demo-content.yaml.j2 \
+        _out/manifests/release/olm/kubevirt-operatorsource.yaml \
+        "_out/manifests/release/olm/bundle/kubevirtoperator.$DOCKER_TAG.clusterserviceversion.yaml" \
+        _out/tests/tests.test \
+        _out/manifests/release/conformance.yaml \
+        _out/templates/manifests/release/conformance.yaml.j2 \
+        _out/manifests/testing/*
+
+    # upload testing manifests to gcs (replaces periodic-kubevirt-update-release-x.y-testing-manifests periodics)
+    gsutil -m rm -r "gs://kubevirt-prow/devel/release/kubevirt/kubevirt/$DOCKER_TAG"
+    gsutil cp -r "_out/manifests/testing gs://kubevirt-prow/devel/release/kubevirt/kubevirt/$DOCKER_TAG/manifests/"
+
+    # generate latest stable version file
+    # will be available under http://storage.googleapis.com/kubevirt-prow/devel/release/kubevirt/kubevirt/stable.txt
+    export latest_stable_release_version="$(
```

Comment:
Shouldn't we just create this file in the postsubmit after all artifacts were updated properly? There we have the tag based on the fact that we build on master and that we build on a tag. Does that make sense?